mysite/chat/models.py

```diff
@@ -110,7 +110,7 @@ def get_spent_time(self):
         :return: datetime.timedelta
         '''
         if not self.last_modify_timestamp:
-            last_msg = self.message_set.all().order_by('-timestamp').first()
+            last_msg = self.message_set.filter(timestamp__isnull=False).order_by('-timestamp').first()
```

Comment:
This looks reasonable. One question though.
Does it make sense to have a message with `timestamp == Null`?